include/bluetooth/mesh/light_ctrl_regulator.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup bt_mesh_light_ctrl_regulator Light Lightness Control Regulator
+ *  @ingroup bt_mesh_light_ctrl
+ *  @{
+ *  @brief Light Lightness Control default regulator API
+ */
+
+#ifndef BT_MESH_LIGHT_CTRL_REGULATOR_H__
+#define BT_MESH_LIGHT_CTRL_REGULATOR_H__
+
+#include <bluetooth/mesh.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Light Lightness Control default regulator context. */
+struct bt_mesh_light_ctrl_spec_regulator {
+	/** @cond INTERNAL_HIDDEN */
+	/** Regulator step timer */
+	struct k_work_delayable timer;
+	/** Internal integral sum. */
+	float i;
+	/** Previous output */
+	uint16_t prev;
+	/** Internal flag field */
+	atomic_t flag;
+	/** Server pointer */
+	struct bt_mesh_light_ctrl_srv *srv;
```

Comment:
When combined with the suggestion about using a "base class" architecture, I think we can remove this pointer and disconnect the regulator from the light ctrl server completely if we remove the call to `bt_mesh_light_ctrl_srv_reg_update`. This can for instance be replaced with a `void (*update)(struct bt_mesh_light_reg *reg, int16_t val)` callback in the regulator structure, that we initialize to point to the standard update function in the INIT macro to maintain the old behavior. 

With this change, we'll also open up for bypassing the light_ctrl_srv altogether, which can fix an issue in the threshold-behavior, which is sort of broken in transitions, as every regulator update will cancel the ongoing transition...

This would allow the regulator to be used for other stuff, unrelated to mesh, which would make it a significantly more valuable module. Arguably, it shouldn't even sit in the Bluetooth mesh subsystem in that case.